assets/policy-extras/hornet.lua

```diff
@@ -0,0 +1,172 @@
+--   extraparams = array of:
+--     addheaders = true/false   -- Add result headers directly to the message?
+--     mode = string            -- make this smtpout (default) to indicate outbound scanning, smtpin for inbound
+--     x_sanitize = true/false  -- make this true to have the engine remove dangerous attachments
+--   msg = message object
+-- returns:
+--   state and score (int) - The spam state and spam score returned for the message. For some states, the
+--      score may provide additional information about the category the message belongs to.
+--      For a complete list of states and scores, please refer to Evaluating State, Score and Verdict.
+
+--   verdict (str) - The message verdict (with no further detail), allowing you to trigger message
+--      rejection without further investigating the state and score if not needed. For a
+--      complete list of verdicts, please refer to Evaluating State, Score and Verdict.
+
+--   spamcause (str) - The full encoded message spam cause.
+
+--   "-" (str) - The encoded envelope information (if provided) that were passed to the library
+--      for scanning (e.g. Inet, Helo, etc.).
+
+--   sanitizeStatus - Status of the sanitize operation:
+--[[  • 0: Successfully removed attachments from the original email.
+      • -1: Error parsing the message: we could not remove attachments.
+      • -2: Malicious attachments may still be present in the modified message.
+      ]]
+--
+--  sanitizedEmail  In case attachments were removed from the original message, the sanitizedEmail
+--      field contains the base64 encoded payload of the modified message.
+
+--  internal (str) - A JSON array containing extra data returned by the filter, including the split
+--[[      recipients and other metadata depending on the options passed to the filter through
+      the AddParams configuration entry.
+      The -2 code may be returned in case the message contained a malicious
+      attachment which could be removed because of the list
+      of SanitizeKeepMime content-types to preserve in the REST
+      service configuration; e.g. an EICAR string contained in a text/html part.
+      It may contain the matched phishing URL or domain (if any), within
+      the InsightUrl=http://www.thisisaphishingurl.com"
+      or InsightDomain=thisisaphishingurl.com".
+      ]]
+--
+
+--  Events (array) - The list of current events that matched the message. Typically ["COVID-19", "EMOTET", …].
+
+--  elapsed (str) - Time it took for the server (in ms) to process the message and return the verdict
+
+function mod:scan(params, extraparams, msg)
+  if not params then
+    print 'No connection table provided'
+    return false
+  end
+
+  if not msg then
+    return false
+  end
+
+  -- Flatten message data to a string
+  local msgdata = msg:get_data()
+
+  -- decompose msg here into vars to pass
+  local modifymsg = extraparams.addheaders or false -- Add result headers directly to the message?
```

Comment:
I'd also collapse `extraparams` into `params` just to keep things simpler with a single table style parameter to the scan function.